tests/integration_tests/flows/test_mariadb.py

```diff
@@ -299,63 +302,50 @@ def test_7_delete_predictor_by_delete_statement(self):
         models = [x['name'] for x in self.mdb.get_models()]
         self.assertTrue(TEST_PREDICTOR_NAME not in models)
 
-        print(f'Test predictor table not exists')
+        print('Test predictor table not exists')
         mindsdb_tables = query('show tables from mindsdb')
         mindsdb_tables = [x[0] for x in mindsdb_tables]
         self.assertTrue(TEST_PREDICTOR_NAME not in mindsdb_tables)
 
 
-def wait_mysql(timeout):
-    global config
-    m = DatabaseWrapper(config)
-
-    start_time = time.time()
-
-    connected = m.check_connections()['default_mariadb']
-    while not connected and (time.time() - start_time) < timeout:
-        time.sleep(2)
-        connected = m.check_connections()['default_mariadb']
-
-    return connected
-
 def stop_mariadb():
     maria_sp = subprocess.Popen(
         ['./cli.sh', 'mariadb-stop'],
-        cwd=pathlib.Path(__file__).parent.absolute().joinpath('../docker/').resolve(),
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL
+        cwd=TESTS_ROOT.joinpath('docker/').resolve(),
+        stdout=OUTPUT,
+        stderr=OUTPUT
     )
     maria_sp.wait()
 
-if __name__ == "__main__":
-    temp_config_path = prepare_config(config)
 
-    maria_sp = subprocess.Popen(
-        ['./cli.sh', 'mariadb'],
-        cwd=pathlib.Path(__file__).parent.absolute().joinpath('../docker/').resolve(),
-        stdout=subprocess.DEVNULL,
-        stderr=subprocess.DEVNULL
-    )
-    atexit.register(stop_mariadb)
-    maria_ready = wait_mysql(START_TIMEOUT)
+if __name__ == "__main__":
+    temp_config_path = prepare_config(config, 'default_mariadb')
+
+    if is_container_run('mariadb-test') is False:
+        subprocess.Popen(
+            ['./cli.sh', 'mariadb'],
+            cwd=TESTS_ROOT.joinpath('docker/').resolve(),
+            stdout=OUTPUT,
+            stderr=OUTPUT
+        )
+        atexit.register(stop_mariadb)
+    maria_ready = wait_db(config, 'default_mariadb')
 
     if maria_ready:
         sp = subprocess.Popen(
             ['python3', '-m', 'mindsdb', '--api', 'mysql', '--config', temp_config_path],
-            stdout=subprocess.DEVNULL,
-            stderr=subprocess.DEVNULL
+            stdout=OUTPUT,
+            stderr=OUTPUT
         )
         atexit.register(sp.kill)
-        port_num = config['api']['mysql']['port']
-    api_ready = maria_ready and wait_port(port_num, START_TIMEOUT)
+
+    api_ready = maria_ready and wait_api_ready(config)
 
     try:
         if maria_ready is False or api_ready is False:
             print(f'Failed by timeout. MariaDB started={maria_ready}, MindsDB started={api_ready}')
             raise Exception()
         unittest.main(failfast=True)
-        print('Tests passed !')
+        print('Tests passed!')
     except Exception as e:
-        print('Tests Failed !')
-        print(e)
-    print('done')
+        print(f'Tests Failed!\n{e}')
```

Comment:
Could we move this logic to the setup method of the class, similarly to how we do it for clickhouse ?

I know it's not much of an advantage, but a bit cleaner and will allow for running with pytest later if we end up switching.

Reply:
ok, will move